sympy/sets/sets.py

```diff
@@ -1427,13 +1427,23 @@ def _contains(self, other):
         return And(*[set.contains(other) for set in self.args])
 
     def __iter__(self):
+        no_iter = True
         for s in self.args:
             if s.is_iterable:
+                no_iter = False
                 other_sets = set(self.args) - set((s,))
                 other = Intersection(other_sets, evaluate=False)
-                return (x for x in s if x in other)
+                for x in s:
+                    c = other.contains(x)
+                    if c == True:
```

Comment:
> we do an audit

I did a grep for places where `contains` is followed by an `==` and changed those. I found a few others by looking for where ` = ` is followed by `contains` to see if a variable was used to store the result and then checked to see how it was used. I included timing info in the commit message.

If there are no objections and tests pass I will commit in about 24 h.